server/svix-server/src/worker.rs

```diff
@@ -284,17 +374,46 @@ async fn dispatch(
         headers
     };
 
-    let client = reqwest::Client::builder()
-        .redirect(reqwest::redirect::Policy::none())
-        .build()
-        .expect("Invalid reqwest Client configuration");
-    let res = client
-        .post(&endp.url)
-        .headers(headers)
-        .timeout(Duration::from_secs(cfg.worker_request_timeout as u64))
-        .json(&payload)
-        .send()
-        .await;
+    // Creates the `HttpConnector` with overridden DNS resoluteon and a connection timeout
+    let mut connector = hyper::client::connect::HttpConnector::new_with_resolver(DnsResolver {
+        allowed_private_subnets: vec![],
+    });
+    connector.set_connect_timeout(Some(Duration::from_secs(cfg.worker_request_timeout as u64)));
+
+    // Creates a new `Connector` type which wraps the `HttpConnector` for TLS support
+    let connector = hyper_rustls::HttpsConnectorBuilder::new()
+        .with_native_roots()
+        .https_or_http()
+        .enable_http1()
+        .enable_http2()
+        .wrap_connector(connector);
+
+    // Actually creates the HTTP client with the above wrapper `Connector`
+    let client: hyper::client::Client<_, hyper::body::Body> =
+        hyper::client::Client::builder().build(connector);
+
+    // Creates a request for the client
+    let mut req = hyper::Request::builder()
+        .method(hyper::Method::POST)
+        .uri(&endp.url);
+
+    // `headers_mut` returns `None` on invalid config, so I'm just extending the mutable
+    // reference with the configured header map so no defaults are overridden
+    req.headers_mut().map(|header_map| {
+        header_map.extend(headers.into_iter());
+        header_map.append(
+            "Content-Type",
+            http::HeaderValue::from_static("application/json"),
+        );
+    });
+
+    let req = req
```

Comment:
Make sure you also set content-length (it's possible hyper will do it for you, but need to make sure).